tests/migrations/test_graph.py

```diff
@@ -255,26 +240,30 @@ def test_validate_consistency(self):
         """
         Tests for missing nodes, using `validate_consistency()` to raise the error.
         """
-        # Build graph
+        # Case 1: missing parent node
         graph = MigrationGraph()
         graph.add_node(("app_a", "0001"), None)
-        # Add dependency with missing parent node (skipping validation).
-        graph.add_dependency("app_a.0001", ("app_a", "0001"), ("app_b", "0002"), skip_validation=True)
-        msg = "Migration app_a.0001 dependencies reference nonexistent parent node ('app_b', '0002')"
+        graph.add_dependency("app_a.0001", ("app_a", "0001"), ("app_a", "0002"), skip_validation=True)
+        msg = "Migration app_a.0001 dependencies reference nonexistent parent node ('app_a', '0002')"
         with self.assertRaisesMessage(NodeNotFoundError, msg):
             graph.validate_consistency()
-        # Add missing parent node and ensure `validate_consistency()` no longer raises error.
-        graph.add_node(("app_b", "0002"), None)
-        graph.validate_consistency()
-        # Add dependency with missing child node (skipping validation).
-        graph.add_dependency("app_a.0002", ("app_a", "0002"), ("app_a", "0001"), skip_validation=True)
```

Comment:
Could you please keep the cross-app reference that we had before.

Reply:
Done